### PR TITLE
Adding logic to fetch memory util with free command.

### PR DESCRIPTION
**Pull request by ShrmaRajat**
Signed-off-by: Rajat Sharma rajat.sharma12@ibm.com

### CODE REVIEW

**Comment by ShrmaRajat** [comment]
@jichenjc @bjhuangr @SeanHQF  PTAL

**Comment by SeanHQF** [comment]
> icic-issue: https://github.ibm.com/zvc/planning/issues/14905

Don't put the issue link here because it's the public site. Put your PR link in the issue instead.

**Comment by SeanHQF** [comment]
@ShrmaRajat have you fetched the latest code to change?

**Comment by SeanHQF** [comment]
@jichenjc @bjhuangr pls help review, thx

**Comment by ShrmaRajat** [comment]
> @ShrmaRajat have you fetched the latest code to change?

yes

File: zvmsdk/monitor.py [hunk 1]
```diff
@@ -135,6 +148,17 @@ def _update_cpumem_data(self, uid_list):
         if self._cache_enabled():
             rdata = self._smtclient.system_image_performance_query(
                 self._namelist)
+            for inst in uid_list:
```
Comment (SeanHQF) [inline zvmsdk/monitor.py:151]: Pls add comment here to explain why we need the below changes. Like "system_image_performance_query can't get the expected memory usage as 'free -h'" etc

File: zvmsdk/monitor.py [hunk 2]
```diff
@@ -74,7 +81,13 @@ def inspect_stats(self, uid_list):
                     'used_mem_kb': used_mem,
                     'max_mem_kb': max_mem,
                     'min_mem_kb': min_mem,
-                    'shared_mem_kb': shared_mem
+                    'shared_mem_kb': shared_mem,
```
Comment (SeanHQF) [inline zvmsdk/monitor.py:84]: Also add comment here.

File: zvmsdk/monitor.py [hunk 3]
```diff
@@ -62,6 +62,13 @@ def inspect_stats(self, uid_list):
                     min_mem = int(user_data['min_memory'].partition(' ')[0])
                     shared_mem = int(
                         user_data['shared_memory'].partition(' ')[0])
+                    used_memory = int(user_data['used_mem'])
```
Comment (SeanHQF) [inline zvmsdk/monitor.py:65]: and comment here.

File: zvmsdk/monitor.py [hunk 4]
```diff
@@ -135,6 +151,18 @@ def _update_cpumem_data(self, uid_list):
         if self._cache_enabled():
             rdata = self._smtclient.system_image_performance_query(
                 self._namelist)
+            # system_image_performance_query can't get the expected memory usage as 'free' command
+            for inst in uid_list:
+                 mem = self._smtclient.execute_cmd(inst, 'free --kilo')
```
Comment (jichenjc) [inline zvmsdk/monitor.py:156]: is it possible something wrong in execute this command?

**Comment by ShrmaRajat** [comment]
@jichenjc @SeanHQF updated PR with suggested approach, PTAL.

File: zvmsdk/monitor.py [hunk 5]
```diff
@@ -135,6 +145,14 @@ def _update_cpumem_data(self, uid_list):
         if self._cache_enabled():
             rdata = self._smtclient.system_image_performance_query(
                 self._namelist)
+            # system_image_performance_query does not return correct memory usage.
+            # Get memory usage from /proc/meminfo
+            for inst in uid_list:
+                 mem = self._smtclient.execute_cmd(inst, 'cat /proc/meminfo')
```
Comment (SeanHQF) [inline zvmsdk/monitor.py:151]: @ShrmaRajat pls add exception handling for this cmd per @jichenjc 's comment

**Review by SeanHQF — APPROVED** [review]

**Review by bjhuangr — APPROVED** [review]

**Comment by bjhuangr** [comment]
please fix the pep8 error and address comments from Robin

**Comment by ShrmaRajat** [comment]
fixed UTs.
in pep8, I see this failure which not related to my changes.
./zvmconnector/restclient.py:1102:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`

**Comment by ShrmaRajat** [comment]
> please fix the pep8 error and address comments from Robin

added exception as per Robin's comment

**Comment by SeanHQF** [comment]
pls update it with the latest code of master, thx

**Review by bjhuangr — APPROVED** [review]

**Pull request merged**